src/Microsoft.AspNet.Mvc.ViewFeatures/ViewComponents/DefaultViewComponentHelper.cs

```diff
@@ -237,7 +249,23 @@ private ViewComponentDescriptor SelectComponent(Type componentType)
                     Resources.FormatViewComponent_IViewComponentFactory_ReturnedNull(descriptor.Type.FullName));
             }
 
+            WriteTelemetryIfEnabled("Microsoft.AspNet.Mvc.BeforeViewComponent", context);
```

Comment:
I don't see much benefit, and I'd personally just lean towards us removing the sync version - which may still happen before release.